cardano-lib/default.nix

```diff
@@ -23,6 +23,15 @@ let
 
   defaultLogConfig = import ./generic-log-config.nix;
   defaultExplorerLogConfig = import ./explorer-log-config.nix;
+  mkExplorerConfig = name: networkConfig: lib.filterAttrs (k: v: v != null) {
+    NetworkName = name;
+    Protocol = networkConfig.Protocol;
+    RequiresNetworkMagic = networkConfig.RequiresNetworkMagic;
+    ByronGenesisFile = networkConfig.ByronGenesisFile or null;
+    ByronGenesisHash = networkConfig.ByronGenesisHash or null;
+    ShelleyGenesisFile = networkConfig.ShelleyGenesisFile or null;
+    ShelleyGenesisHash = networkConfig.ShelleyGenesisHash or null;
+  };
```

Comment:
for the keys where the name matches up, you could `inherit (networkConfig) Protocol;`